src/box/sql.c

```diff
@@ -1693,3 +1673,45 @@ sql_foreign_key_create(const char *name, uint32_t child_id, uint32_t parent_id,
 	region_truncate(region, used);
 	return rc;
 }
+
+int
+sql_check_create(const char *name, uint32_t space_id, uint32_t func_id,
+		 uint32_t fieldno, bool is_field_ck)
+{
+	const struct space *space = space_by_id(space_id);
+	if (space == NULL) {
+		diag_set(ClientError, ER_NO_SUCH_SPACE, space_name(space));
+		return -1;
+	}
+	struct tuple_constraint_def *cdefs;
+	uint32_t count;
+	char buf[64];
+	char value[16];
+	const char *path;
+	mp_encode_uint(value, func_id);
+
+	if (is_field_ck) {
+		count = space->def->fields[fieldno].constraint_count;
+		cdefs = space->def->fields[fieldno].constraint_def;
+		int len = snprintf(buf, 64, "format[%u].constraint",
+				   fieldno + 1);
+		assert(len > 0 && len < 64);
+		(void)len;
+		path = buf;
+	} else {
+		count = space->def->opts.constraint_count;
+		cdefs = space->def->opts.constraint_def;
+		path = "flags.constraint";
+	}
+	for (uint32_t i = 0; i < count; ++i) {
+		if (cdefs[i].type != CONSTR_FUNC)
+			continue;
+		if (strcmp(name, cdefs[i].name) == 0) {
+			diag_set(ClientError, ER_CONSTRAINT_EXISTS, "CHECK",
+				 name, space_name(space));
+			return -1;
+		}
+	}
+	int rc = sql_constraint_create(name, space_id, path, value);
+	return rc;
```

Comment:
Apparently you can also do `return sql_constraint_create(...)` without `rc` variable.

Reply:
True, fixed.